lib/pure/complex.nim

```diff
@@ -10,75 +10,126 @@
 
 
 ## This module implements complex numbers.
+## Complex numbers are currently implemented as generic on a 64-bit or 32-bit float
 {.push checks:off, line_dir:off, stack_trace:off, debugger:off.}
 # the user does not want to trace a part
 # of the standard library!
 
 
+
 import
   math
 
-const
-  EPS = 1.0e-7 ## Epsilon used for float comparisons.
-
 type
-  Complex* = tuple[re, im: float]
-    ## a complex number, consisting of a real and an imaginary part
+  Complex*[T] = tuple[re, im: T]
+  ## a complex number, consisting of a real and an imaginary part
+  Complex128* = tuple[re, im: float64]
+  Complex64* = tuple[re, im: float32]
```

Comment:
Isnt it better to define `Complex128` and `Complex64` as `Complex[float64]` and `Complex[float32]`?